src/Tezos/Transaction.h

```diff
@@ -0,0 +1,44 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "Address.h"
+#include <boost/multiprecision/cpp_int.hpp>
+
+namespace TW {
+namespace Tezos {
+
+class Transaction {
+public:
+    typedef boost::multiprecision::uint256_t uint256_t;
+
+    uint256_t nonce;
```

Comment:
Here's a canonical JSON representation of a transaction in the Tezos protocol:  
{
	"amount": "1", /// Amount to send as an integer value. I.E. "1" represents .000001 XTZ
	"counter": "30738", // Address Counter, should be +1 over existing value
	"destination": "tz3WXYtyDUNL91qfiCJtVUX746QpNv5i5ve5", // Address receiving
	"fee": "1272", // Fee for operation - should always be the same 
	"gas_limit": "10100", // Gas limit - should always be the same
	"kind": "transaction", // Hardcoded value representing operation kind
	"source": "tz1XVJ8bZUXs7r5NV8dHvuiBhzECvLRLR3jW", // Address making transaction
	"storage_limit": "257" // Gas limit - should always be the same
}

I need to figure out how to to map these values to a hex encoded string.. will update here when I know more.

Reply:
Here's another test case: 

That transaction above should encode to: 
da8eb4f57f98a647588b47d29483d1edfdbec1428c11609cee0da6e0f27cfc3808000081faa75f741ef614b0e35fcc8c90dfa3b0b95721f80992f001f44e81020100026fde46af0356a0476dae4e4600172dc9309b3aa400

(But I don't know how :))

Reply:
Let branch = BMNY6Jkas7BzKb7wDLCFoQ4YxfYoieU7Xmo1ED3Y9Lo3ZvVGdgW

Base58CheckDecode branch and remove first two bytes (should be [1, 52]). Based on branch above this should be:
[218, 142, 180, 245, 127, 152, 166, 71, 88, 139, 71, 210, 148, 131, 209, 237, 253, 190, 193, 66, 140, 17, 96, 156, 238, 13, 166, 224, 242, 124, 252, 56] -> 
[1, 52, 218, 142, 180, 245, 127, 152, 166, 71, 88, 139, 71, 210, 148, 131, 209, 237, 253, 190, 193, 66, 140, 17, 96, 156, 238, 13, 166, 224, 242, 124, 252, 56]